include/tunnler/room.h

```diff
@@ -0,0 +1,121 @@
+// Copyright 2017 Tunnler Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+//
+
+#pragma once
+
+#include <atomic>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "tunnler/tunnler.h"
+
+#include "RakPeerInterface.h"
+
+const uint16_t DefaultRoomPort = 1234;
+
+// This is what a server [person creating a server] would use.
+class Room final {
+public:
+    enum class State {
+        Open,   // The room is open and ready to accept connections.
+        Closed, // The room is not opened and can not accept connections.
+    };
+
+    struct Member {
+        std::string nickname; ///< The nickname of the member.
+        MacAddress mac_address; ///< The assigned mac address of the member.
+        RakNet::AddressOrGUID network_address; ///< The network address of the remote peer.
+    };
+
+    using MemberList = std::vector<Member>;
```

Comment:
I didn't read the code yet [just your review comments], but introducing "Server" would be even more confusing.
The general concept of naming these functions was that a non-network person or even layman would get these terms - this will also be reflected by the GUI.